lib/io/adc.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include <cstdio>
+
+namespace hyped::io {
+class Adc {
+ public:
+  /**
+   * @param pin
+   */
+  explicit Adc(const uint32_t pin);
```

Comment:
What purpose does the `explicit` server here?

Reply:
Thought this was to prevent implicit conversions to uint32_t. Don't know if we really need to enforce that but kept it jic

Reply:
I don't believe this is what `explicit` does.

This is how I understand it:

Say you have a function

```cpp
void f(const Adc adc) { /* ... */ }
```

and you call it like 

```cpp
const uint32_t pin = 8;
f(pin);
```

this won't work if your constructor is `explicit`. Rather, you have to do 

```cpp
f(Adc(pin));
```

I would consider the former bad practice but I don't see any reason to enforce it explicitly. (Mainly because it adds to confusions such as yours.)